main.go

```diff
@@ -0,0 +1,123 @@
+/*
```

Comment:
I think we can move this file to `cmd/main.go` within Dockerfile in the future.
What do you think @Jeffwan ?

Reply:
Sounds good. `cmd/training-operator.v1/main.go`

Reply:
Hi Andrey, I include this change in a separate PR. This PR includes most changes we made in last two stages. There's follow up PR coming to address these suggestions.